docs/integrations/creating-nodes/build/reference/ui-elements.md

```diff
@@ -732,4 +732,72 @@ Display a yellow box with a hint or extra info. Refer to [Node UI design](/integ
   default: '',
 },
 ```
-![Notice](/_images/integrations/creating-nodes/notice.png)
\ No newline at end of file
+![Notice](/_images/integrations/creating-nodes/notice.png)
+
+## Hints
+
+There are two types of hints: input hints and larger hints.
+
+Input hints are small lines of text below a user input field.
+
+Larger hints are a more powerful and flexible option than [Notice](#notice). Use them to display longer hints, in the input panel, output panel, or node details view. 
+
+### Add an input hint
+
+Add the `hint` parameter to the field:
+
+```ts
+{
+	displayName: 'URL',
+	name: 'url',
+	type: 'string',
+	hint: 'Enter a URL',
+	...
+}
+```
+
+### Add a larger hint
+
+```ts
+description: INodeTypeDescription = {
+	...
+	hints: [
+		{
+			// The hint message. [TODO: can you use HTML or markdown in the text?]
+			message: "This node has many input items. Consider enabling <b>Execute Once</b> in node\'s settings.",
+			// Choose from: info, warning, danger. Changes the color.
+			// [TODO: which colors? In the screenshots all is grey]
+			type: 'info',
+			// Choose from: inputPane, outputPane, ndv [TODO: defaults to? Or is this required?]
+			location: 'outputPane',
+			// Choose from: always, beforeExecution, afterExecution. [TODO: Defaults to always?]
+			whenToDisplay: 'beforeExecution',
+			// Optional. An expression. If it resolves to true, n8n displays the message. Defaults to true.
+			displayCondition: '={{ $parameter["operation"] === "select" && $input.all().length > 1 }}'
+		}
+	]
+	...
+}
+```
+
+### Add a dynamic hint to a programmatic-style node
+
+In programmatic-style nodes you can create a dynamic message that includes information from the node output. [TODO: confirm you can't include dynamic data in declarative style]
```

Comment:
you could, e.g. if display condition tied to some parameter in declarative hints it would be updated if it changes
main difference of programmatic style that node run data is available, custom logic could be used, and that they could be shown only after execution